src/scripts/analyze-migration-diff.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Analyzes migration schema snapshots to show high-level changes between versions.
+ * Provides a concise summary of tables added, removed, and modified.
+ */
+
+import crypto from 'crypto'
+import fs from 'fs'
+import path from 'path'
+
+interface TableFieldDiff {
+  fieldsAdded: string[]
+  fieldsRemoved: string[]
+  fieldsModified: string[]
+}
+
+interface DiffAnalysis {
+  migrationName: string
+  tablesAdded: string[]
+  tablesRemoved: string[]
+  tablesModified: Map<string, TableFieldDiff>
+}
+
+interface MigrationSnapshot {
+  version: string
+  dialect: string
+  tables: Record<string, TableSchema>
+}
+
+interface TableSchema {
+  name: string
+  columns: Record<string, unknown>
+  indexes?: Record<string, unknown>
+  foreignKeys?: Record<string, unknown>
+  compositePrimaryKeys?: Record<string, unknown>
+  uniqueConstraints?: Record<string, unknown>
+  checkConstraints?: Record<string, unknown>
+}
+
+function hashObject(obj: unknown): string {
+  return crypto.createHash('sha256').update(JSON.stringify(obj)).digest('hex')
+}
+
+function diffTableFields(oldTable: TableSchema, newTable: TableSchema): TableFieldDiff {
+  const oldColumns = new Set(Object.keys(oldTable.columns))
+  const newColumns = new Set(Object.keys(newTable.columns))
+
+  const fieldsAdded: string[] = []
+  const fieldsRemoved: string[] = []
+  const fieldsModified: string[] = []
+
+  // Find added columns
+  for (const column of newColumns) {
```

Comment:
You have two `Set()` - you could just use symmetric difference methods for these